Mods/src/PuppiMod.cc

```diff
@@ -186,7 +190,35 @@ PuppiMod::Process()
 
   auto* pv = vertexes->At(0);
 
-  auto* pfCandidates = GetObject<PFCandidateCol>(fPFCandidatesName);
+  PFCandidateArr *InvertedParticles = new PFCandidateArr;
+  InvertedParticles->SetName(fInvertedParticlesName);
+
+  auto* inPfCandidates = GetObject<PFCandidateCol>(fPFCandidatesName);
+  PFCandidateCol* pfCandidates = NULL;
+  PFCandidateArr* leptonCandidates = NULL;
```

Comment:
leptonCandidates does not have to be an Array, I suppose? (Array is an actual storage with memory allocation and object copying - as far as I see you use the lepton candidates as constant input objects, so PFCandidateOArr should be enough). Also it does not need to be a pointer - new and delete are expensive operations, so I'd suggest something like

```
PFCandidateOArr leptonCandidates;
...
leptonCandidates.Add(cand);
```